lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1224,6 +1224,42 @@ def group_log_entries_by_tag_and_local_resource_id(chunk)
     # collection of entries.
     def determine_group_level_monitored_resource_and_labels(tag,
                                                             local_resource_id)
+      # Determine the monitored resource based on the local_resource_id.
+      # Different monitored resource types have unique ids in different format.
+      # We will query Metadata Agent for the monitored resource. Return the
+      # legacy monitored resource (either the instance resource or the resource
+      # inferred from the tag) if failed to get a monitored resource from
+      # Metadata Agent with this key.
+      #
+      # Examples:
+      # "container.<container_id>" // Docker container.
+      # "k8s_pod.<namespace_name>.<pod_name>" // GKE pod.
+      if local_resource_id
+        resource = monitored_resource_from_local_resource_id(local_resource_id)
+        if resource
+          resource.freeze
```

Comment:
Would it make sense for `monitored_resource_from_local_resource_id` to return the resource already frozen?